Assets/_Laz/Scripts/Environment/Puzzles/Connection/Vertex.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+
+namespace Laz
+{
+    public class Vertex
+    {
+        private bool _isActivated = false;
+        private Node _frontNode = null;
+        private Node _backNode = null;
+
+        public event Action OnVertexCompleted;
+
+        public Vertex(Node frontNode, Node backNode)
```

Comment:
A bit confused about the definition of a Vertex. The constructor takes a frontNode and a backNode so it kinda feels more like an 'Edge' to me. What's the difference between Node and Vertex?

Reply:
Oh yeah i'll use the word edge. 

An edge is where two faces meet. A vertex is a corner where edges meet. The plural is vertices. - Google

Reply:
Yeah I think Edge would be a good name, either that or 'Segment' or something like that Just cause Vertex and Node are very similar concepts

Reply:
I'm sticking with edge since there's a chance i need to reuse it for other puzzles with Graph Theory, and they're normally called nodes and edges.

Reply:
Sounds good